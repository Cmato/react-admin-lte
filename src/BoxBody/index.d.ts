export { default } from './BoxBody';
export * from './BoxBody';
