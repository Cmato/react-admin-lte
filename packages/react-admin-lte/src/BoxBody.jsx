import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

const propTypes = {
  noPadding: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.node,
};

const defaultProps = {
  noPadding: false,
};

const BoxBody = ({
  className,
  children,
  noPadding,
}) => {
  const classes = {
    'box-body': true,
    'no-padding': noPadding,
  };

  return (
    <div className={classNames(className, classes)}>
      {children}
    </div>
  );
};

BoxBody.propTypes = propTypes;
BoxBody.defaultProps = defaultProps;

export default BoxBody;
