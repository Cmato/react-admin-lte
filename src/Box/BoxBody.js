import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const BoxBody = ({
  children,
  className: classNameProp,
  noPadding,
  ...other
}) => {
  const className = classNames({
    'box-body': true,
    'no-padding': noPadding
  }, classNameProp);

  return (
    <div className={className} {...other}>
      {children}
    </div>
  );
};

BoxBody.propTypes = {
  /**
   * The contents of the box body.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * If `true`, padding will be removed from the box body.
   */
  noPadding: PropTypes.bool
};

BoxBody.defaultProps = {
  noPadding: false
};

export default BoxBody;
