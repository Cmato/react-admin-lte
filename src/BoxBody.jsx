import React from 'react';
import classNames from 'classnames';

const propTypes = {
  padding: React.PropTypes.bool,
};

const defaultProps = {
  padding: true,
};

const BoxBody = ({
  className,
  children,
  padding,
}) => {
  const classes = {
    'box-body': true,
    'no-padding': !padding,
  };

  return (
    <div className={classNames(className, classes)}>
      {children}
    </div>
  );
};

BoxBody.propTypes = propTypes;
BoxBody.defaultProps = defaultProps;

export default BoxBody;
