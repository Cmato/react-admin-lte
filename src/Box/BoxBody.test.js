import React from 'react';
import { shallow } from 'enzyme';
import BoxBody from './BoxBody';

it('renders default', () => {
  expect(shallow(<BoxBody><div /></BoxBody>))
    .toMatchSnapshot();
});

it('renders default', () => {
  expect(shallow(<BoxBody noPadding><div /></BoxBody>))
    .toMatchSnapshot();
});
