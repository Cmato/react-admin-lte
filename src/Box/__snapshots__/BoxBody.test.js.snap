// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
<div
  className="box-body"
>
  <div />
</div>
`;

exports[`renders default 2`] = `
<div
  className="box-body no-padding"
>
  <div />
</div>
`;
